go-client/client/fieldmanager.go

```diff
@@ -324,7 +361,11 @@ func (fm *fieldManager) FetchTablesRepeating(ctx context.Context, appServiceClie
 // Once Close has been called, no other methods should be used on the fieldManager (except Close again, which will do nothing).
 func (fm *fieldManager) Close() {
 	if fm.chanClose != nil {
+		// This waits for the executor to stop.
+		closeConfirm := make(chan struct{})
+		fm.chanClose <- reqClose{out: closeConfirm}
 		close(fm.chanClose)
```

Comment:
Yes, it does. Thankfully, it can use the same lock that all of the other `Close` methods use: the one in `Client`. I added a note explaining that the lock should be held when calling the function, same as all the other stub `Close` methods.